feature-libs/product-configurator/rulebased/cpq/cpq-configurator-overview-normalizer.spec.ts

```diff
@@ -204,12 +204,23 @@ describe('CpqConfiguratorOverviewNormalizer', () => {
   it('should map user input as attribute value', () => {
     attr.userInput = 'input';
     attr.displayAs = Cpq.DisplayAs.INPUT;
+    attr.dataType = Cpq.DataType.INPUT_STRING;
     const ovAttrs = serviceUnderTest['convertAttribute'](attr, CURRENCY);
     expect(ovAttrs.length).toBe(1);
     expect(ovAttrs[0].value).toEqual('input');
     expect(ovAttrs[0].productCode).toBeUndefined();
   });
 
+  it('should map user input with not supported DataType as NOT_IMPLEMENTED', () => {
+    attr.userInput = 'input';
+    attr.displayAs = Cpq.DisplayAs.INPUT;
+    attr.dataType = Cpq.DataType.INPUT_NUMBER;
+    const ovAttrs = serviceUnderTest['convertAttribute'](attr, CURRENCY);
+    expect(ovAttrs.length).toBe(1);
+    expect(ovAttrs[0].value).toEqual('NOT_IMPLEMENTED');
+    expect(ovAttrs[0].productCode).toBeUndefined();
+  });
+
```

Comment:
this does not seem to be related to 'move value from input field'. did you fix multiple issues with this PR?

Reply:
Yes. Several issues (directly or indirectly related to handling of input fields) are fixed with this pull request. All adjustments are mentioned in the pull request comment.